inc/mailcollector.class.php

```diff
@@ -550,17 +554,24 @@ function deleteOrImportSeveralEmails($emails_ids = [], $action = 0, $entity = 0)
             $collector->fetch_emails = 0;
             //Connect to the Mail Box
             $collector->connect();
-            // Get Total Number of Unread Email in mail box
-            $tot = $collector->getTotalMails(); //Total Mails in Inbox Return integer value
 
-            for ($i=1; $i<=$tot; $i++) {
-               $uid = imap_uid($collector->marubox, $i);
-               $head = $collector->getHeaders($uid);
+            foreach ($collector->storage as $uid => $message) {
+               if ($collector->fetch_emails >= $collector->maxfetch_emails) {
```

Comment:
I am not sure this limit is relevant. Indeed, we are searching for specific emails here. If they are not in the limit of `maxfetch_emails`, they will not be reimported.